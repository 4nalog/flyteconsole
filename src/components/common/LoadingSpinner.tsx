import * as classnames from 'classnames';
import * as React from 'react';

import { CircularProgress } from '@material-ui/core';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
    container: {
        alignItems: 'center',
        display: 'flex',
        justifyContent: 'center',
        '&.large': { height: '300px' },
        '&.medium': { height: '80px' },
        '&.small': { height: '40px' }
    }
});

type SizeValue = 'small' | 'medium' | 'large';
interface LoadingSpinnerProps {
    size?: SizeValue;
}

const spinnerSizes: Record<SizeValue, number> = {
    small: 24,
    medium: 48,
    large: 96
};

/** Renders a loading spinner. Size options are 'small', 'medium', and 'large' */
export const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
    size = 'large'
}) => {
    const styles = useStyles();
    return (
        <div className={classnames(styles.container, size)}>
            <CircularProgress size={spinnerSizes[size]} />
        </div>
    );
};
