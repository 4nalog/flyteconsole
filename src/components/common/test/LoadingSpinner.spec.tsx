import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { LoadingSpinner } from '../LoadingSpinner';

describe('LoadingSpinner', () => {
    it('renders large size by default', () => {
        const noProp = renderer.create(<LoadingSpinner />).toJSON();
        const large = renderer.create(<LoadingSpinner size="large" />).toJSON();
        expect(noProp).toEqual(large);
    });

    it('renders small size', () => {
        expect(
            renderer.create(<LoadingSpinner size="small" />).toJSON()
        ).toMatchSnapshot();
    });

    it('renders medium size', () => {
        expect(
            renderer.create(<LoadingSpinner size="medium" />).toJSON()
        ).toMatchSnapshot();
    });

    it('renders large size', () => {
        expect(
            renderer.create(<LoadingSpinner size="large" />).toJSON()
        ).toMatchSnapshot();
    });
});
