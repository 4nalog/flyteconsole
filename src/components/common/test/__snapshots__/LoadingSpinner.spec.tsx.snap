// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders large size 1`] = `
<div
  className="makeStyles-container-1 large"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`LoadingSpinner renders medium size 1`] = `
<div
  className="makeStyles-container-1 medium"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 48,
        "width": 48,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`LoadingSpinner renders small size 1`] = `
<div
  className="makeStyles-container-1 small"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
